src/types/Common.ts

```diff
@@ -26,3 +26,25 @@ export type TimeInSeconds = number;
 
 export type RefreshIntervalInMs = TimeInMilliseconds;
 export type DurationInSeconds = TimeInSeconds;
+export type FromToInMilliseconds = {
+  from: TimeInMilliseconds;
+  to?: TimeInMilliseconds;
+};
+
+export type TimeRange = DurationInSeconds | FromToInMilliseconds;
```

Comment:
I guess FromTo could be convenient for describing a time range in the past, but conceptually it's the same as queryTime + duration, which is what we use now in our APIs and which works naturally with Prometheus.  So why change to To and From parameters?  I could be missing something but it seems like TimeRange could just be duration and optional queryTime (default=now).   Helper functions could still convert to FromToInMilliseconds if necessary.  As is maybe it's over-complicated.

Either way, is there a benefit to FromToInMilliseconds as opposed to FromToInSeconds?   Seconds seems to be used more often in the API (maybe not charts, I'm not sure).

Reply:
You're probably correct, I didn't realize I could have the same with what we already have.
As about ms versus seconds: I find on the contrary more stuff being in milliseconds, for instance queryTime is in ms, no? DurationInSeconds being more the anomaly .. anyway the question isn't relevant anymore if I remove this FromTo thing.
I'll do that after the break

Reply:
@jshaughn I think the best is to use the existing http API with duration/queryTime as you said, but still have a specific "from-to" model like the one I suggest because it maps 1-to-1 with the actual component that we're building (a "from" field and a "to" field) so the app state is more accurately described.

To take an example that shows a behavioural difference: if "from" is defined but "to" is undefined, if this is accurately described in the app state, "from" will stay fixed upon refreshes whereas "to" will keep pointing to "now" (which is I think a valid scenario that users would expect). On the contrary, if it gets converted to a duration in the app state, the initial "from" that the user explicitly selected would be lost after the first refresh.

For this kind of things I think it's better to have the model accurately describing the UI component. Conversions to duration would then be performed only when fetching data (ie. in a stateless process)

Reply:
@jotak I'd like to have a meeting with you to show you what I've done in My PR for graph replay.  We're in an area of overlap with, I think, slightly different goals to support custom time range for charts and custom time range for the graph.   We should talk it out to make sure we come up with something that satisfies all of the foreseeable Kiali needs.  Of course any interested person is invited as well.  I'll contact you...

Reply:
By the way, today I decided to use TimeInMilliseconds and IntervalInMilliseconds for basically all of the client-side time manipulations in graph (in my PR).  Leaving TimeInSeconds only for the rest api for use with queryTime param to the backend API, which expects seconds because that works well with Prom (which actually uses time.Time).  I still like seconds for the queryTime param but our API is not yet supported, so if there is a movement to use millis in the rest api we could still make that change.